commands/cli.js

```diff
@@ -88,31 +88,17 @@ function redisCLI (uri, client) {
   client.on('data', function (data) {
     reply.execute(data)
   })
-  client.on('error', function (error) {
-    cli.error(error)
-    process.exit(1)
-  })
-  client.on('end', function () {
-    console.log('\nDisconnected from instance.')
-    process.exit(0)
+  return new Promise((resolve, reject) => {
+    client.on('error', reject)
+    client.on('end', function () {
+      console.log('\nDisconnected from instance.')
+      resolve()
+    })
   })
 }
 
-function match (config, lookup) {
-  for (var key in config) {
-    if (lookup.test(key)) {
-      return config[key]
-    }
-  }
-  return null
-}
-
-function maybeTunnel (redis, config) {
-  let bastions = match(config, /_BASTIONS/)
-  let hobby = redis.plan.indexOf('hobby') === 0
-  let uri = url.parse(redis.resource_url)
-
-  if (bastions != null) {
+function bastionConnect ({uri, bastions, config}) {
+  return new Promise((resolve, reject) => {
     let tunnel = new Client()
     tunnel.on('ready', function () {
```

Comment:
@dickeyxxx you need to handle an error here since if we never get a `ready` event I think this will hang indefinitely